profiles/common/modules/custom/tmgmt_poetry/inc/tmgmt_poetry.webservice.inc

```diff
@@ -0,0 +1,470 @@
+    'access callback' => TRUE,
+  );
+  $items[drupal_get_path("module", "tmgmt_poetry") . "/wsdl/PoetryIntegration.wsdl"] = array(
+    'page callback' => '_tmgmt_poetry_wsdl',
+    'access callback' => TRUE,
+  );
+  return $items;
+}
+
+
+/**
+ * Function to define our webservice and register the available methods.
+ */
+function _tmgmt_poetry_service_callback() {
+  // Generate our own SOAP server.
+  $context = stream_context_create();
+  $uri = url(drupal_get_path("module", "tmgmt_poetry") . "/wsdl/PoetryIntegration.wsdl", array('absolute' => TRUE, 'language' => (object) array('language' => FALSE)));
+
+  $server = new SoapServer($uri, array(
+    'stream_context' => $context,
+    'cache_wsdl' => WSDL_CACHE_NONE,
+  ));
+  $server->AddFunction("FPFISPoetryIntegrationRequest");
+  $server->handle();
+}
+
+/**
+ * Helper function to generate an answer xml for received Poetry requests.
+ */
+function _tmgmt_poetry_generate_answer_xml($job = NULL, $message = 'OK', $code = 0, $received_request = NULL) {
+  // Create initial XML element using POETRY headers.
+  $xml = new SimpleXMLElement("<?xml version=\"1.0\" encoding=\"utf-8\" ?>
+    <POETRY xmlns:xsi=\"http://www.w3.org/2001/XMLSchema-instance\"
+    xsi:noNamespaceSchemaLocation=\"http://intragate.ec.europa.eu/DGT/poetry_services/poetry.xsd\">
+    </POETRY>");
+
+  // Add main request element.
+  $request = $xml->addChild('request');
+
+  if ($received_request !== NULL) {
+    $reference = implode('/', (array) $received_request->demandeId);
+  }
+  elseif ($job !== NULL) {
+    $reference = $job->reference;
+  }
+
+  $request->addAttribute('id', $reference);
+  $request->addAttribute('type', 'status');
+
+  // Add request identifier.
+  $demande_id_data = explode('/', $reference);
+  $demande_id = $request->addChild('demandeId');
+  $demande_id->addChild('codeDemandeur', $demande_id_data[0]);
+  $demande_id->addChild('annee', $demande_id_data[1]);
+  $demande_id->addChild('numero', $demande_id_data[2]);
+  $demande_id->addChild('version', $demande_id_data[3]);
+  $demande_id->addChild('partie', $demande_id_data[4]);
+  $demande_id->addChild('produit', $demande_id_data[5]);
+
+  $status = $request->addChild('status');
+  $status->addAttribute('type', 'request');
+  $status->addAttribute('code', $code);
+
+  $status_date = $status->addChild('statusDate', date('d/m/Y'));
+  $status_date->addAttribute('format', 'dd/mm/yyyy');
+
+  $status_time = $status->addChild('statusTime', date('h:i:s'));
+  $status_time->addAttribute('format', 'hh:mm:ss');
+
+  $status->addChild('statusMessage', $message);
+
+  return $xml;
+}
+
+/**
+ * Helper function to authenticate request made to our service.
+ */
+function _tmgmt_poetry_authenticate($user, $password) {
+  // Get Poetry configuration.
+  $poetry = variable_get("poetry_service");
+  if (($poetry['callback_user'] == $user) &&
+      ($poetry['callback_password'] == $password)) {
+    return TRUE;
+  }
+  return FALSE;
+}
```

Comment:
This is overly verbose. This is equivalent to the following:

```
return $poetry['callback_user'] == $user && $poetry['callback_password'] == $password;
```

These parameters are not documented but the look like strings. If they are strings then you should always do a literal comparison using `===` instead of using `==` which allows casting. The input types are not checked, so using the latter will have unintended consequences in case other types are passed. For example if `$user` equals `TRUE` then this will return `TRUE`.